### PR TITLE
HT-25 : Create more tests for `ActivityLogModel`

### PR DESCRIPTION
**Pull request by pranjal5399**

### CODE REVIEW

File: packages/api/source/models/ActivityLogModel.ts [hunk 1]
```diff
@@ -19,6 +19,10 @@ const activityLogSchema = new Schema(
         context: {
             type: Object,
         },
+        createdAt: {
+            type: Date,
+            immutable: true,
+        },
```
Comment (itssamuelrowe) [inline packages/api/source/models/ActivityLogModel.ts:25]: Why did you add this field?

File: packages/api/test/models/activityLog.test.ts [hunk 2]
```diff
@@ -89,4 +89,88 @@ describe("ActivityLog Model", function () {
...
+            component: "no-api",
+        });
+
+        await assertThrowsAsync(
+            async () => newLog.save(),
+            "The component attribute is not from defined origins.",
+        );
+    });
+
+    it("should run when context is null", async () => {
```
Comment (itssamuelrowe) [inline packages/api/test/models/activityLog.test.ts:141]: What do you mean by "should run"? It should be "should be created". Follow the same convention everywhere.

File: packages/api/test/models/activityLog.test.ts [hunk 3]
```diff
@@ -89,4 +89,88 @@ describe("ActivityLog Model", function () {
...
+            context: null,
+        });
+        await newLog.save();
+        assert.isFalse(
+            newLog.isNew,
+            "The log should be persisted to the database.",
+        );
+    });
+
+    it("should run when context is undefined", async () => {
```
Comment (itssamuelrowe) [inline packages/api/test/models/activityLog.test.ts:153]: "should be created"

File: packages/api/test/models/activityLog.test.ts [hunk 4]
```diff
@@ -89,4 +89,88 @@ describe("ActivityLog Model", function () {
...
+        );
+    });
+
+    it("should run if createdAt is valid", async () => {
+        const newLog = new ActivityLogModel({ ...logTemplate });
+        await newLog.save();
+        assert.typeOf(newLog.createdAt, "date", "This is a valid date");
+    });
+
+    it("should run if updatedAt is valid", async () => {
```
Comment (itssamuelrowe) [inline packages/api/test/models/activityLog.test.ts:171]: The title doesn't make any sense.

File: packages/api/test/models/activityLog.test.ts [hunk 5]
```diff
@@ -89,4 +89,88 @@ describe("ActivityLog Model", function () {
...
+            context: undefined,
+        });
+        await newLog.save();
+        assert.isFalse(
+            newLog.isNew,
+            "The log should be persisted to the database.",
+        );
+    });
+
+    it("should run if createdAt is valid", async () => {
```
Comment (itssamuelrowe) [inline packages/api/test/models/activityLog.test.ts:165]: The title doesn't make sense.

**Review by itssamuelrowe — CHANGES_REQUESTED** [review]

File: packages/api/test/models/activityLog.test.ts [hunk 6]
```diff
@@ -89,4 +89,88 @@ describe("ActivityLog Model", function () {
...
+        );
+    });
+
+    it("should be created if createdAt is valid", async () => {
+        const newLog = new ActivityLogModel({ ...logTemplate });
+        await newLog.save();
+        assert.typeOf(newLog.createdAt, "date", "This is a valid date");
+    });
+
+    it("should be created if updatedAt is valid", async () => {
```
Comment (itssamuelrowe) [inline packages/api/test/models/activityLog.test.ts:171]: "should be created with a default value for `updatedAt`"

File: packages/api/test/models/activityLog.test.ts [hunk 7]
```diff
@@ -89,4 +89,88 @@ describe("ActivityLog Model", function () {
...
+        const newLog = new ActivityLogModel({ ...logTemplate });
+        await newLog.save();
+        assert.typeOf(newLog.updatedAt, "date", "This is a valid date");
```
Comment (itssamuelrowe) [inline packages/api/test/models/activityLog.test.ts:174]: "This is a valid date"

The message that you specify to an assertion should say what went wrong. Read the docs.

The correct message is:

"The updatedAt attribute should be assigned by default."

File: packages/api/test/models/activityLog.test.ts [hunk 8]
```diff
@@ -89,4 +89,88 @@ describe("ActivityLog Model", function () {
...
+        assert.isFalse(
+            newLog.isNew,
+            "The log should be persisted to the database.",
+        );
+    });
+
+    it("should be created if createdAt is valid", async () => {
+        const newLog = new ActivityLogModel({ ...logTemplate });
+        await newLog.save();
+        assert.typeOf(newLog.createdAt, "date", "This is a valid date");
```
Comment (itssamuelrowe) [inline packages/api/test/models/activityLog.test.ts:168]: "This is a valid date"

The message that you specify to an assertion should say what went wrong. Read the docs.

The correct message is:

"The createdAt attribute should be assigned by default."

File: packages/api/source/models/ActivityLogModel.ts (hunk 1 above)
Comment (itssamuelrowe) [inline packages/api/source/models/ActivityLogModel.ts:25]: Why doesn't this have a default value?

File: packages/api/test/models/activityLog.test.ts [hunk 9]
```diff
@@ -89,4 +89,88 @@ describe("ActivityLog Model", function () {
...
+            context: undefined,
+        });
+        await newLog.save();
+        assert.isFalse(
+            newLog.isNew,
+            "The log should be persisted to the database.",
+        );
+    });
+
+    it("should be created if createdAt is valid", async () => {
```
Comment (itssamuelrowe) [inline packages/api/test/models/activityLog.test.ts:165]: "should be created with a default value for `createdAt`"

**Review by itssamuelrowe — CHANGES_REQUESTED** [review]

File: packages/api/source/models/ActivityLogModel.ts (hunk 1 above)
Comment (pranjal5399) [reply packages/api/source/models/ActivityLogModel.ts:25]: It will be automatically added as {timestamps:true}

**Review by itssamuelrowe — APPROVED** [review]
